### PR TITLE
better parser for step call

### PR DESCRIPTION
**Pull request by TonioSnowden**
Works best for :
```python
from notte.env import NotteEnv
import litellm

goal = "Find best flights from Boston to Tokyo on google flights"
messages = [{"role": "system", "content": goal}]
max_steps = 10

def llm_agent(messages):
    response = litellm.completion(
        model="cerebras/llama3.1-70b",
        messages=messages,
    )
    return response.choices[0].message.content

async def main(max_steps=10):
    async with NotteEnv(headless=False) as env:
        while '<done>' not in messages[-1]['content'] and max_steps > 0:
            resp = llm_agent(messages) # query your llm policy.
            obs = await env.chat(resp) # query notte with llm response.
            messages.append({"role": "assistant", "content": resp})
            messages.append({"role": "user", "content": obs})
            max_steps -= 1

await main()
```

### CODE REVIEW

File: notte/common/parser.py [hunk 1]
```diff
@@ -111,8 +110,8 @@ def textify(self, obs: Observation) -> str:
 Use the exact following format:
 <action>
 {
-"action-id": {} # if an action has no params
-"action-id": { "name":"value" } # for action with params
+"action-id": "<YOUR_ACTION_ID>",
+"params": { "param_name":"param_value" }
```
Comment (giordano-lucas) [inline notte/common/parser.py:114]: probably better to write "<YOUR_PARAM_NAME>": "YOUR_PARAM_VALUE"

File: notte/common/parser.py [hunk 2]
```diff
@@ -86,8 +85,8 @@ def step(self, text: str) -> EnvStepParams:
         if action is None or not isinstance(action, str):
             raise ValueError("No action found")
         d = json.loads(action)
-        action_id = list(d.keys())[0]
-        params = d[action_id] if d[action_id] != {} else None
+        action_id = d["action-id"]
```
Comment (giordano-lucas) [inline notte/common/parser.py:88]: @TonioSnowden can you raise a value error if these keys are not present ?

File: notte/env.py [hunk 3]
```diff
@@ -147,7 +147,7 @@ async def chat(self, text: str) -> str:
             return self._parser.textify(obs)
         elif endpoint == "step":
             step_params = self._parser.step(text)
-            obs = await self.step(step_params.action_id, step_params.params, enter=False)
+            obs = await self.step(step_params.action_id, step_params.params, enter=True)
```
Comment (giordano-lucas) [inline notte/env.py:150]: I don't think that this is needed

**Review by giordano-lucas — CHANGES_REQUESTED** [review]

**Comment by andreakiro** [comment]
+Precommits are failing, install and fix them before we merge
```
pre-commit install
```

**Review by andreakiro — CHANGES_REQUESTED** [review]
+Precommits are failing, install and fix them before we merge
```
pre-commit install
```

File: notte/env.py (hunk 3 above)
Comment (TonioSnowden) [reply notte/env.py:150]: For me needed otherwise it doesn't succeed to insert something (like confirm to put the departure city)

Comment (giordano-lucas) [reply notte/env.py:150]: I think it's none by default ```
async def step(
        self,
        action_id: str,
        params: dict[str, str] | str | None = None,
        enter: bool | None = None,
    )
   ```

**Review by andreakiro — APPROVED** [review]

**Pull request merged**